frontend/src/components/shared/StreamlitMarkdown/StreamlitMarkdown.tsx

```diff
@@ -303,9 +303,32 @@ export function LinkWithTargetBlank(props: LinkProps): ReactElement {
   )
 }
 
+function fixBodyForHeader(tag: string, body: string): string {
+  switch (tag) {
+    case "h1": {
+      body = `# ${body}`
```

Comment:
nit: Can probably just do 
```javascript
return `# ${body}`
```

This removes the need for `break` statements and cleans it up. As a result, you can just make the default outside the switch statement.